modules/openapi-generator/src/main/resources/aspnetcore/2.1/Startup.3_0.mustache

```diff
@@ -0,0 +1,150 @@
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="configuration"></param>
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        /// <summary>
+            /// The application configuration.
+        /// </summary>
+        public IConfiguration Configuration { get; }
+
+        /// <summary>
+        /// This method gets called by the runtime. Use this method to add services to the container.
+        /// </summary>
+        /// <param name="services"></param>
+        public void ConfigureServices(IServiceCollection services)
+        {
+{{#authMethods}}
+{{#isApiKey}}
+{{#-first}}
+            services.AddTransient<IAuthorizationHandler, ApiKeyRequirementHandler>();
+            services.AddAuthorization(authConfig =>
+            {
+{{/-first}}            
+                authConfig.AddPolicy("{{name}}",
+                    policyBuilder => policyBuilder
+                        .AddRequirements(new ApiKeyRequirement(new[] { "my-secret-key" },"{{name}}")));
+{{#-last}}
+            });
+{{/-last}}
+{{/isApiKey}}
+{{/authMethods}}
+
+            // Add framework services.
+            services
+                .AddControllers()
+                .SetCompatibilityVersion(CompatibilityVersion.Version_3_0)
+                .AddNewtonsoftJson(opts =>
+                    {
+                        opts.SerializerSettings.ContractResolver = new CamelCasePropertyNamesContractResolver();
+                        opts.SerializerSettings.Converters.Add(new StringEnumConverter
+                        {
+                            NamingStrategy = new CamelCaseNamingStrategy()
+                        });
+                    }
+                );
+
+            {{#useSwashbuckle}}
+
+            services
+                .AddSwaggerGen(c =>
+                {
+                    c.SwaggerDoc("{{#version}}{{{version}}}{{/version}}{{^version}}v1{{/version}}", new OpenApiInfo
+                    {
+                        Version = "{{#version}}{{{version}}}{{/version}}{{^version}}v1{{/version}}",
+                        Title = "{{#appName}}{{{appName}}}{{/appName}}{{^appName}}{{packageName}}{{/appName}}",
+                        Description = "{{#appName}}{{{appName}}}{{/appName}}{{^appName}}{{packageName}}{{/appName}} (ASP.NET Core 3.0)",
+                        Contact = new OpenApiContact()
+                        {
+                           Name = "{{#infoName}}{{{infoName}}}{{/infoName}}{{^infoName}}OpenAPI-Generator Contributors{{/infoName}}",
+                           Url = new Uri("{{#infoUrl}}{{{infoUrl}}}{{/infoUrl}}{{^infoUrl}}https://github.com/openapitools/openapi-generator{{/infoUrl}}"),
+                           Email = "{{#infoEmail}}{{{infoEmail}}}{{/infoEmail}}"
+                        },
+                        TermsOfService = "{{#termsOfService}}{{{termsOfService}}}{{/termsOfService}}"
```

Comment:
This is now a URI and we need to provide a link to the TOS. 
Needs commenting out for now